unity-renderer/Assets/Scripts/MainScripts/DCL/WorldRuntime/SceneBoundariesController/SceneBoundsChecker.cs

```diff
@@ -385,13 +398,10 @@ private void SetEntityCollidersInsideBoundariesState(MeshesInfo meshesInfo, bool
 
         private void SetComponentsInsideBoundariesValidState(IDCLEntity entity, bool isInsideBoundaries)
         {
-            if(entity.isInsideSceneBoundaries == isInsideBoundaries || !DataStore.i.sceneBoundariesChecker.componentsCheckSceneBoundaries.ContainsKey(entity.entityId))
+            if (entity.isInsideSceneBoundaries == isInsideBoundaries || !DataStore.i.sceneBoundariesChecker.componentsCheckSceneBoundaries.ContainsKey(entity.entityId))
                 return;
 
-            foreach (IOutOfSceneBoundariesHandler component in DataStore.i.sceneBoundariesChecker.componentsCheckSceneBoundaries[entity.entityId])
-            {
-                component.UpdateOutOfBoundariesState(isInsideBoundaries);
-            }
+            foreach (IOutOfSceneBoundariesHandler component in DataStore.i.sceneBoundariesChecker.componentsCheckSceneBoundaries[entity.entityId]) { component.UpdateOutOfBoundariesState(isInsideBoundaries); }
```

Comment:
I just noticed this. `DataStore.i` does a lookup in a dictionary, and this is a hot path. We should cache it.